metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/bulk/BulkDocumentWriter.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.elasticsearch.bulk;
+
+import org.apache.metron.indexing.dao.update.Document;
+
+import java.util.List;
+
+/**
+ * Writes documents to an index in bulk.
+ *
+ * <p>Partial failures within a batch can be handled individually by registering
+ * a {@link FailureListener}.
+ *
+ * @param <D> The type of document to write.
+ */
+public interface BulkDocumentWriter<D extends Document> {
+
+    /**
+     * A listener that is notified when a set of documents have been
+     * written successfully.
+     * @param <D> The type of document to write.
+     */
+    interface SuccessListener<D extends Document> {
```

Comment:
How's this compare with the recent changes for decoupling the Parser Bolts from parser exec? At a glance, this seems very similar from an abstraction standpoint. Any reason we would want callbacks here as compared with the decisions we made for the parser bolt being more direct?

https://github.com/apache/metron/pull/1213#issuecomment-427086826

Reply:
Yep, I think you're right.  Wish I would have noticed this before. Daft.

Reply:
Done. See latest commit.